lms/djangoapps/course_api/blocks/api.py

```diff
@@ -43,24 +45,21 @@ def get_blocks(
         return_type (string): Possible values are 'dict' or 'list'. Indicates
             the format for returning the blocks.
     """
-    # construct BlocksAPITransformer
-    blocks_api_transformer = BlocksAPITransformer(
-        block_counts,
-        student_view_data,
-        depth,
-        nav_depth
-    )
-
-    # list of transformers to apply, adding user-specific ones if user is provided
-    transformers = [blocks_api_transformer]
+    # create ordered list of transformers, adding BlocksAPITransformer at end.
+    transformers = BlockStructureTransformers()
     if user is not None:
         transformers += COURSE_BLOCK_ACCESS_TRANSFORMERS + [ProctoredExamTransformer()]
+    transformers += [
+        BlocksAPITransformer(
+            block_counts,
+            student_view_data,
+            depth,
+            nav_depth
+        )
+    ]
```

Comment:
I don't have a strong opinion on this (I don't even know what the shape of super-user usage_info might be). But I think it's reasonable to have Transformers understand `None` as an anonymous user. That way there can be a simple view to transformer list mapping, which would make it easier to plug in future ones without edx-platform proper having to know any of the details.